src/services/redisSessionStorage.ts

```diff
@@ -317,6 +295,51 @@ export default class RedisSessionStorage extends RedisStorageUtils
     );
   }
 
+  /*
+  * Return the promises needed to store session info and 
+  * update session info set
+  */
+  private saveSessionInfo(
+    user: User
+  ): ReadonlyArray<Promise<Either<Error, boolean>>> {
+    const newSessionInfo: SessionInfo = {
+      createdAt: new Date(),
+      sessionToken: user.session_token
+    };
+    const sessionInfoKey = `${sessionInfoKeyPrefix}${user.session_token}`;
+    const saveSessionInfo = new Promise<Either<Error, boolean>>(resolve => {
+      this.redisClient.set(
+        sessionInfoKey,
+        JSON.stringify(newSessionInfo),
+        "EX",
+        this.tokenDurationSecs,
+        (err, response) =>
+          resolve(
+            this.falsyResponseToError(
+              this.singleStringReply(err, response),
+              new Error("Error setting user token info")
+            )
+          )
+      );
+    });
+    const updateSessionInfoSet = new Promise<Either<Error, boolean>>(
+      resolve => {
+        this.redisClient.sadd(
+          `${userSessionsSetKeyPrefix}${user.fiscal_code}`,
+          sessionInfoKey,
+          (err, response) =>
+            resolve(
+              this.falsyResponseToError(
+                this.integerReply(err, response),
+                new Error("Error updating user tokens info set")
+              )
+            )
+        );
+      }
+    );
+    return [saveSessionInfo, updateSessionInfoSet];
```

Comment:
if you return multiple promises the caller needs to know how to handle them (Promise.all? Promise.race? how to handle one rejection vs multiple rejections?) -> this leaks the abstraction of a single operation

I suggest you return a single `Promise<Either<Error, boolean>>`

Reply:
fixed in f5f264b